File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n == 0) {
            return 0;
        }
        
        if (n == 1 || n == 2) {
            return 1;
        }
        
        std::vector<int> res (n + 1, 0);
        res[1] = 1,
        res[2] = 1;

        
        for (int i = 3; i <= n; i++) {
            res[i] = res[i - 1] + res[i - 2] + res[i - 3];
        }
        
        return res[n];
    }
};

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.rs
impl Solution {
    pub fn tribonacci(n: i32) -> i32 {
        if n == 0 {
            return 0;
        }
        
        if n < 3 {
            return 1;
        }
        
        let (mut t1, mut t2, mut t3, mut res) = (0, 1, 1, 2);
        
        for i in 2..n {
            res = t1 + t2 + t3;
            (t1, t2, t3) = (t2, t3, res);
        }
        
        res
    }
}